slick/src/main/scala/slick/util/InternalArrayQueue.scala

```diff
@@ -0,0 +1,175 @@
+package slick.util
+
+import java.util
+
+private[util] final class InternalArrayQueue[E >: Null <: AnyRef](capacity: Int) {
```

Comment:
Yes it's a copy of the old `ManagedArrayBlockingQueue`. So, same license applies.

Maybe I should add the old comment again?

/** A simplified copy of `java.util.concurrent.ArrayBlockingQueue` with additional logic for
  * temporarily rejecting elements based on the current size. All features of the original
  * ArrayBlockingQueue have been ported, except the mutation methods of the iterator. See
  * `java.util.concurrent.ArrayBlockingQueue` for documentation. */